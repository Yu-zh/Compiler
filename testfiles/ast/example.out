Program
  List
    Func
      IntType
      IdDecl ID="gcd1"
      List
        Para
          IntType
          IdDecl ID="a"
        Para
          IntType
          IdDecl ID="b"
      List
        WhileStmt
          Neq
            IdUse ID="a"
            IdUse ID="b"
          List
            IfStmt
              Gt
                IdUse ID="a"
                IdUse ID="b"
              List
                Assign
                  IdUse ID="a"
                  Min
                    IdUse ID="a"
                    IdUse ID="b"
              List
                Assign
                  IdUse ID="b"
                  Min
                    IdUse ID="b"
                    IdUse ID="a"
        Return
          Opt
            IdUse ID="a"
    Func
      IntType
      IdDecl ID="gcd2"
      List
        Para
          IntType
          IdDecl ID="a"
        Para
          IntType
          IdDecl ID="b"
      List
        IfStmt
          Eq
            IdUse ID="b"
            Numeral NUMERAL="0"
          List
            Return
              Opt
                IdUse ID="a"
          List
        Return
          Opt
            Call
              IdUse ID="gcd2"
              List
                IdUse ID="b"
                Mod
                  IdUse ID="a"
                  IdUse ID="b"
    Func
      IntType
      IdDecl ID="main"
      List
      List
        VarDecl
          IntType
          IdDecl ID="a"
          Opt
        VarDecl
          IntType
          IdDecl ID="b"
          Opt
        Assign
          IdUse ID="a"
          Call
            IdUse ID="read"
            List
        Assign
          IdUse ID="b"
          Call
            IdUse ID="read"
            List
        FuncCallStmt
          Call
            IdUse ID="print"
            List
              Call
                IdUse ID="gcd1"
                List
                  IdUse ID="a"
                  IdUse ID="b"
        FuncCallStmt
          Call
            IdUse ID="print"
            List
              Call
                IdUse ID="gcd2"
                List
                  IdUse ID="a"
                  IdUse ID="b"
        VarDecl
          IntType
          IdDecl ID="diff"
          Opt
            Min
              Call
                IdUse ID="gcd1"
                List
                  IdUse ID="a"
                  IdUse ID="b"
              Call
                IdUse ID="gcd2"
                List
                  IdUse ID="a"
                  IdUse ID="b"
        FuncCallStmt
          Call
            IdUse ID="print"
            List
              IdUse ID="diff"
        Return
          Opt
            Numeral NUMERAL="0"
