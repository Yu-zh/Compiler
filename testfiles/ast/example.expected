Program
  List
    Func
      IdDecl ID="gcd1"
      List
        Para
          IdDecl ID="a"
        Para
          IdDecl ID="b"
      List
        WhileStmt
          Neq
            IdUse ID="a"
            IdUse ID="b"
          List
            IfStmt
              Gt
                IdUse ID="a"
                IdUse ID="b"
              List
                Assign
                  IdUse ID="a"
                  Min
                    IdUse ID="a"
                    IdUse ID="b"
              List
                Assign
                  IdUse ID="b"
                  Min
                    IdUse ID="b"
                    IdUse ID="a"
        Return
          IdUse ID="a"
    Func
      IdDecl ID="gcd2"
      List
        Para
          IdDecl ID="a"
        Para
          IdDecl ID="b"
      List
        IfStmt
          Eq
            IdUse ID="b"
            Numeral NUMERAL="0"
          List
            Return
              IdUse ID="a"
          List
        Return
          Call
            IdUse ID="gcd2"
            List
              IdUse ID="b"
              Mod
                IdUse ID="a"
                IdUse ID="b"
    Func
      IdDecl ID="main"
      List
      List
        VarDecl
          IdDecl ID="a"
          Opt
        VarDecl
          IdDecl ID="b"
          Opt
        Assign
          IdUse ID="a"
          Read
            List
        Assign
          IdUse ID="b"
          Read
            List
        Print
          Call
            IdUse ID="gcd1"
            List
              IdUse ID="a"
              IdUse ID="b"
        Print
          Call
            IdUse ID="gcd2"
            List
              IdUse ID="a"
              IdUse ID="b"
        VarDecl
          IdDecl ID="diff"
          Opt
            Min
              Call
                IdUse ID="gcd1"
                List
                  IdUse ID="a"
                  IdUse ID="b"
              Call
                IdUse ID="gcd2"
                List
                  IdUse ID="a"
                  IdUse ID="b"
        Print
          IdUse ID="diff"
        Return
          Numeral NUMERAL="0"

